### PR TITLE
 also offer to open settings if detecting java fails

### PR DESCRIPTION
**Pull request by adietish**
this follows up on #220

### CODE REVIEW

**Comment by adietish** [comment]
This adds a button to the error dialog which allows you to open the vscode settings. Unfortunately vscode is only capable of presenting the global settings, not the "Servers View" settings. So the user would still have to search for it. 
image
Looking for opinions: @robstryker @jeffmaury better? no benefit?

**Comment by jeffmaury** [comment]
What's this 0 version ? This needs to be improved

**Comment by adietish** [comment]
@jeffmaury the "0" doesn't really matter here, it's on purpose: I used an invalid java executable which is causing this output. Nevertheless we could not print the java version if it's "0".
The more important question is: hitting "Configure Java" will pop up the general settings in VSCode. There's apparently no way to present our settings right away. Good enough? Not worth it?

**Comment by jeffmaury** [comment]
Or display 'Invalid' instead ?

**Comment by adietish** [comment]
@jeffmaury what about the main question here? Do you think that this additional button is helpful?

**Comment by adietish** [comment]
discussed it and we think that offering the button is helpful even though it would be far better if we could open our own setting section, which unfortunately isn't possible in vscode currently.
The other improvement that we discussed is to download the jdk from https://adoptopenjdk.net/

**Comment by adietish** [comment]
updated error text when no java was detected (compare to screenshot above):
image

File: src/requirements.ts [hunk 1]
```diff
@@ -74,8 +74,12 @@ function checkJavaVersion(javaHome: string): Promise<number> {
       const javaExecutable = path.resolve(javaHome, 'bin', JAVA_FILENAME);
       cp.execFile(javaExecutable, ['-version'], {}, (error, stdout, stderr) => {
             const javaVersion = parseMajorVersion(stderr);
-            if (javaVersion < 8) {
-                rejectWithDownloadUrl(reject, `Java 8 or newer is required to run this extension. Java ${javaVersion} was found at ${javaHome}. Please download and install a recent JDK`);
+            if (!javaVersion) {
+                rejectWithDownloadUrl(reject, `Java 8 or newer is required. No Java was found on your system..
+                Please get a recent JDK or configure if for "Servers View" if it already exists`);
```
Comment (robstryker) [inline src/requirements.ts]: Is this a typo? "Please get a recent JDK or configure if for..."   ?   "if for" ? 

Same for next else-if as well.

Comment (adietish) [reply src/requirements.ts]: @robstryker fixed.

**Comment by tellison** [comment]
Anything we can do to make this easier, or help this along from the AdoptOpenJDK project side?

We [publish builds](https://adoptopenjdk.net/releases.html) as images, installers or simple archives, and there is [an API](https://api.adoptopenjdk.net/) for getting the release info and download links, e.g. [the latest mac Java 11](https://api.adoptopenjdk.net/v2/latestAssets/releases/openjdk11?os=mac&arch=x64&type=jdk&openjdk_impl=hotspot).

**Comment by adietish** [comment]
@tellison sorry for the late reply, this went somehow forgotten. Thanks a lot for the pointers! Guess that for the current task we're just fine but there's plenty of room for improvements (ex. choose the jdk to download etc.).

**Comment by adietish** [comment]
@lstocchi can you please review this orphaned PR, please? I just rebased and pushed it, hope I got the merging right.

**Comment by lstocchi** [comment]
@adietish added the configure java btn 
image

The problem is that it's not possible to open the settings by code so we need to tell the user to do it. My solution is when the user click on "Configure Java" to open the quickpick with the "Open Settings" already typed + added an info message to tell the user what to do.
image
image

It seems that the extension doesn't have access to the workspace settings if a workspace exists so we can only work with the User settings. I also tried to open the User Settings file by typing the exact path to the openTextDocument cmd without success.

**Comment by lstocchi** [comment]
@adietish found there is a command to open the settings.json programmatically so we're using this one 'workbench.action.openSettingsJson'

**Pull request merged**